DPB_converger.py

```diff
@@ -0,0 +1,363 @@
+    def __init__(self, msg):
+        Exception.__init__(self, msg)
+
+
+def get_loc_files(dir_path):
+    """get localization files from directory"""
+    files = []
+    for file in os.listdir(dir_path):
+        if file.endswith(".csv") and "tracked" not in file:
+            files.append(dir_path + "\\" + file)
+    return files
+
+
+def write_statistics(output_path, dataset):
+    """
+    writes the contents of a statistics dictionary to a file
+    """
+    output = open(output_path, 'w', newline='')
+    for i in dataset.keys():
+        output.write(i + ',')
+    output.write('\n')
+    writer = csv.writer(output, delimiter=',')
+    writer.writerows(zip(*dataset.values()))
+    output.close()
+
+
+def get_matching_files(directory, target, exclusion_string):
+    """
+    Search in directory and its subdirectories for files with target in their name but missing the exclusion string
+
+    :param directory: all files in this directory & subdirectories are checked
+    :param target: substring of filename
+    :param exclusion_string: excludes files with this string in the name
+    :return: List of matching file paths
+    """
+    matching_files = []
+    for path, subdirs, files in os.walk(directory):
+        for name in files:
+            if target in name.lower():
+                if exclusion_string not in name.lower():
+                    matching_files.append(os.path.join(path, name))
+    return matching_files
+
+
+def define_swft_command(file_path, tau, exp_displacement, exp_noise_rate, max_displacement, max_displacement_pp,
+                        p_bleach, p_switch, precision, diff_limit):
+    """swft command per file"""
+    file_command = 'swft.exe ' + file_path + ' --tau ' + '"' + tau + '" ' + '--out_values "all" '
+    file_command += '--exp_displacement ' + '"' + exp_displacement + '"' + ' '
+    file_command += '--exp_noise_rate ' + '"' + str(exp_noise_rate) + '"' + ' '
+    file_command += '--max_displacement ' + '"' + max_displacement + '"' + ' '
+    file_command += '--max_displacement_pp ' + '"' + max_displacement_pp + '"' + ' '
+    file_command += '--p_bleach ' + '"' + p_bleach + '"' + ' '
+    file_command += '--p_switch ' + '"' + p_switch + '"' + ' '
+    file_command += '--precision ' + '"' + str(precision) + '"' + ' '
+    file_command += '--diffraction_limit ' + '"' + diff_limit + '"' + ' '
+    return file_command
+
+
+def gather_Data(filepaths):
+    """
+    Collects the displacement and pBleach data of all notebook outputs in the given list of directories
+    """
+    #gathers a list of all relevant files
+    displacement_files = []
+    bleach_files = []
+    for working_directory in filepaths:
+        displacement_files += get_matching_files(working_directory, 'displacement', 'histogram')
+        bleach_files += get_matching_files(working_directory, 'p_bleach', 'histogram')
+    displacement_data = []
+    bleach_data = []
+    # reads the values and puts them into one list, then returns it
```

Comment:
", then returns it" can be deleted and the return clarified in the docstring of the function